frontend/awx/resources/inventories/InventoryForm.tsx

```diff
@@ -1,65 +1,71 @@
 import { PageFormSelectOrganization } from '../../access/organizations/components/PageFormOrganizationSelect';
-import { getOrganizationByName } from '../../access/organizations/utils/getOrganizationByName';
 import { PageFormInstanceGroupSelect } from '../../administration/instance-groups/components/PageFormInstanceGroupSelect';
 import { AwxItemsResponse } from '../../common/AwxItemsResponse';
 import { PageFormLabelSelect } from '../../common/PageFormLabelSelect';
 import { getAddedAndRemoved } from '../../common/util/getAddedAndRemoved';
 import { InstanceGroup } from '../../interfaces/InstanceGroup';
 import { Inventory } from '../../interfaces/Inventory';
 import { Label } from '../../interfaces/Label';
+import { PageFormSection } from '../../../../framework/PageForm/Utils/PageFormSection';
 
-interface InventoryFields extends FieldValues {
-  inventory: Inventory;
-  instanceGroups?: InstanceGroup[];
-  id: number;
+export interface InventoryFields {
+  name: string;
+  description: string;
+  organization: { name: string; id: number };
+  instanceGroups: InstanceGroup[];
+  labels: Label[] | [];
+  variables: string;
+  prevent_instance_group_fallback: boolean;
+}
+export interface InventoryCreate {
+  organization: number;
+  name: string;
+  description: string;
+  variables: string;
+  prevent_instance_group_fallback: boolean;
 }
 
 export function CreateInventory() {
   const { t } = useTranslation();
-  const navigate = useNavigate();
-  const postRequest = usePostRequest<{ id: number }, Inventory>();
+  const pageNavigate = usePageNavigate();
+  const postRequest = usePostRequest<InventoryCreate, Inventory>();
 
   const onSubmit: PageFormSubmitHandler<InventoryFields> = async (values) => {
-    const { inventory, instanceGroups } = values;
-    if (inventory.summary_fields.organization.name) {
-      try {
-        const organization = await getOrganizationByName(
-          inventory.summary_fields.organization.name
-        );
-        if (!organization) throw new Error(t('Organization not found.'));
-        inventory.organization = organization.id;
-      } catch {
-        throw new Error(t('Organization not found.'));
-      }
-    }
-    // Create new inventory
-    const newInventory = await postRequest('/api/v2/inventories/', inventory);
+    const { organization, labels, instanceGroups, ...rest } = values;
+
+    const newInventory = await postRequest('/api/v2/inventories/', {
+      ...rest,
+      organization: organization.id,
+    });
 
     // Update new inventory with selected instance groups
-    await submitInstanceGroups(newInventory, instanceGroups ?? [], []);
+    if (instanceGroups?.length > 0)
+      await submitInstanceGroups(newInventory, instanceGroups ?? [], []);
 
     // Update new inventory with selected labels
-    await submitLabels(newInventory, inventory.summary_fields.labels.results);
+    if (labels.length > 0) await submitLabels(newInventory, labels);
```

Comment:
Re: the E2E failure on editing the inventory: I suspect this has introduced an issues in the PATCH request when we attempt to update labels. The request needs to be patched with `summary_fields.labels`.